ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs

```diff
@@ -100,10 +100,28 @@ data Mempool m blk = Mempool {
       -- index of transaction hashes that have been included on the blockchain.)
       addTxs :: [GenTx blk] -> m [(GenTx blk, ApplyTxErr blk)]
 
-      -- | Get all transactions in the mempool (oldest to newest)
+      -- |
+    , syncState :: STM m ()
```

Comment:
If we want to trace the transactions removed from the mempool, the tracing has to be done *after* `syncState` completed because we can't trace inside `STM`. But in order to do that, we need to know which transactions were actually removed by `syncState`. So should `syncState` return a list of removed transactions? Or do you have another approach in mind?

Reply:
We can do that or there's one alternative: have `syncState` accumulate temporary tracing output into another `TVar` and have the background thread (that also calls the internal equiv of `syncState`) do the tracing.

Reply:
Decided to go with the first option @mrBliss suggested.